### PR TITLE
[Ansor] Support multiple output ops and fix Python API printing

### PR DESCRIPTION
**Pull request by comaniac**
As I'm trying to use Ansor to tune the operators generated from `te.gradient`, I fixed some issues in this PR so that now Ansor can tune the backward ops (regardless the performance). Detail change list:

- [Multiple Output] Create a schedule with all outputs instead of the first output.
- [Python API Printing] Change `tvm.thread_axis` to `te.thread_axis`.
- [Python API Printing] Add prefix to all iterators to avoid name conflict; otherwise we may refer to the wrong iterator with the same name hint (e.g., `ax0`). For example:

```python
ax0, ... = tuple(pad_temp_data_grad.op.axis) + ...
...
ax0, ... = tuple(pad_temp_shared.op.axis) + ...
s[pad_temp_data_grad].split(ax0, factor=4)
```
where `ax0` in `split` should refer to the `ax0` from `pad_temp_data_grad`, but it has been overridden by `pad_temp_shared` after `cache_read`. This PR improves `CleanName` by providing an optional prefix so that we can differentiate those iterators by their stages.

cc @merrymercy @jcf94

### CODE REVIEW

**Comment by comaniac** [comment]
Current issue: `te.InferBound` failed for schedules with multiple outputs. Investigating.

**Comment by comaniac** [comment]
Two more bug fixing:

1. Only specify output tensors when creating a new schedule; otherwise InferBound may crash.

2. Update compute DAG attributes such as access analyzer and ops after layout rewriting (cc @minminsun ).

All auto_scheduler tests are now passed.

File: src/auto_scheduler/search_policy/sketch_policy_rules.h [hunk 1]
```diff
@@ -74,6 +75,8 @@ class SketchGenerationRule {
    */
   virtual std::vector<std::pair<State, int>> Apply(const SketchPolicyNode& policy,
                                                    const State& state, int stage_id) const = 0;
+
+  virtual std::string GetRuleName() const = 0;
```
Comment (merrymercy) [inline src/auto_scheduler/search_policy/sketch_policy_rules.h:79]: Document it

File: src/auto_scheduler/compute_dag.cc [hunk 2]
```diff
@@ -969,6 +969,9 @@ void ComputeDAG::RewriteLayout(const Array<Step>& transform_steps) {
       }
     }  // end for placeholder
   }    // end for stage
+  p_dag->access_analyzer = AccessAnalyzer(p_dag->tensors);
+  p_dag->ops = p_dag->access_analyzer->ops_topo_order;
+  p_dag->flop_ct = FlopEstimator().EstimateFlop(p_dag->ops);
```
Comment (merrymercy) [inline src/auto_scheduler/compute_dag.cc:974]: Why does it work before without the fix?

Comment (comaniac) [reply src/auto_scheduler/compute_dag.cc:974]: I would like to know the answer too... For the ApplySteps in compute DAG, previously we simply put all non-placeholder ops to the list and take the last op (assuming it is the only output) to create a new schedule. After using `access_analyzer.is_output` to check what ops should be put to the list, `is_output` throws exceptions due to the out-of-date `is_output` map. That's why I found this issue. Maybe most use cases so far don't enable layout rewrite?

**Review by merrymercy — APPROVED** [review]

**Comment by tqchen** [comment]
Thanks @comaniac @merrymercy

**Pull request merged**